Roadmap.md

```diff
@@ -0,0 +1,64 @@
+# Roadmap
+
+If you are interested in making a contribution to Podium, this page outlines some changes we are planning to focus on in the near future. Feel free to propose improvements and moficiations either via [discussions](https://github.com/TakeLab/podium/discussions) or by raising an [issue](https://github.com/TakeLab/podium/issues).
+
+Order does not reflect importance.
+
+## Major changes
+
+- Dynamic application of Fields
+  - Right now, for every change in Fields the dataset needs to be reloaded. The goal of this change would be to allow users to replace or update a Field in a Dataset. The Dataset should be aware of this change (e.g. by keeping a hash of the Field object) and if it happens, recompute all the necessary data for that Field.
+
+  The current pattern is:
+  ```python
+    # Load a dataset
+    fields = {'text':text, 'label':label}
+    dataset = load_dataset(fields=fields)
+
+    # Decide to change something with one of the Fields
+    text = Field(..., tokenizer=some_different_tokenizer)
```

Comment:
I know this is an example, but I'd make this something copy-pasteable nonetheless with comments 

text = Field("example_field", tokenizer=str.split('#') # changing tokenizer to split on '#'

Reply:
I kinda intentionally left it abstract/as pseudocode as it would clutter the page too much if it were concrete. I think it's fine this way, it's not intended to be ran.